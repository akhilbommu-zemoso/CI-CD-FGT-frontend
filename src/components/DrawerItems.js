import React from 'react'
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider'
import { Link } from 'react-router-dom'
function DrawerItems(props) {
    return (
        <Link data-testid="link" to={props.link} className={props.linksclass} onClick={() => { props.setSelected(2 ** props.index) }} >
            <ListItem
                selected={(props.selected >> props.index) & 1}
                classes={(props.selected:props.active)}>
                <Divider classes={{ root: props.dividerColor }}
                flexItem={(props.selected >> props.index) & 1}
                orientation="vertical" />
                 <ListItemIcon>{props.icon}</ListItemIcon>
                <ListItemText primary={props.itemName} />
            </ListItem>
        <Divider />
        </Link >
      
    )
}

export default DrawerItems