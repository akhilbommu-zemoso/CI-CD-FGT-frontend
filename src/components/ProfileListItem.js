import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import { Link } from "react-router-dom";
const useStyles = makeStyles((theme) => ({
  linksclass: {
    color: "black",
    paddingTop: "0px",
    textDecoration: "none",
    "&:hover": {
      color: "black",
      textDecoration: "none",
    },
  },
  active: {
    backgroundColor: "rgb(4,136,255)",
    color: "white",
    "&:hover": {
      backgroundColor: "rgb(4,136,255)",
      opacity: 0.9,
    },
  },
  font: {
    fontSize: "14px",
  },
}));
export default function ProfileListItem(props) {
  const classes = useStyles();
  return (
    <div>
      <div>
        <Link className={classes.linksclass}>
          <ListItem
            data-testid="listitem"
            button={(props.selected >> props.index) & 1}
            classes={{ button: props.active }}
            onClick={() => {
              props.setSelected(2 ** props.index);
            }}
          >
            <ListItemText
              classes={{ primary: classes.font }}
              primary={props.itemName}
            />
          </ListItem>
        </Link>
      </div>
    </div>
  );
}
