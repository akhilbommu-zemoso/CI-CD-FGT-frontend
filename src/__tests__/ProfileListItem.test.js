import React, { Component } from 'react'
import { render, fireEvent, getAllByText, getByText, getByTestId } from '@testing-library/react'
import { renderHook, act } from '@testing-library/react-hooks'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from "react-test-renderer";
import DrawerItems from '../components/DrawerItems';
import { createMemoryHistory } from "history";
import ProfileListItem from '../components/ProfileListItem';
Enzyme.configure({ adapter: new Adapter() });
const setSelected = jest.fn();
it('should take a snapshot', () => {
    const { asFragment } = render(
        <Router>
            <ProfileListItem />
        </Router>
    )
    expect(asFragment(<Router><ProfileListItem itemName="Profile"/></Router>)).toMatchSnapshot()
});
it('visit /profile on click', () => {
    const { asFragment,getByTestId ,getByText} = render(
        <Router>
            <ProfileListItem
                itemName="Profile"
                link="profile"
                setSelected={setSelected}
                index={1}
            />
        </Router>
    )
    fireEvent.click(getByTestId("listitem"))
    expect(setSelected).toHaveBeenCalled
    expect(getByText("Profile")).toBeInTheDocument
});