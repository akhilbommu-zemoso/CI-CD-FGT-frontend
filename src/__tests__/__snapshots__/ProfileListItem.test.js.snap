// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <a
        class="makeStyles-linksclass-1"
        href=""
      >
        <li
          class="MuiListItem-root MuiListItem-gutters"
          data-testid="listitem"
        >
          <div
            class="MuiListItemText-root"
          />
        </li>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
